src/fu-engine.c

```diff
@@ -3488,7 +3491,7 @@ fu_engine_load (FuEngine *self, GError **error)
 
 	/* add devices */
 	fu_engine_plugins_setup (self);
-	fu_engine_plugins_coldplug (self);
+	fu_engine_plugins_coldplug (self, TRUE);
```

Comment:
Hmm, why no longer allow coldplug to be called during coldplug?  Do you think that's the root of ending up with two devices?  I'll need to check and see if this works correctly for the flows I expect.

Reply:
The root problem was the device list was accepting two different FuDevice's with the same ID. When we recoldplug it's pointless going to all the trouble of creating a FuDevice (e.g. the amt one is slooow) just for the device-list to throw it away, so I figured opting-in to the recodplug was a good idea. The TRUE/FALSE passed to fu_engine_plugins_coldplug just signifies if the call to coldplug is "original" i.e. as part of the daemon startup, or as a recoldplug.